userspace/chisel/chisel.cpp

```diff
@@ -1074,84 +1075,43 @@ struct filename
     string ext;
 };
 
-static filename split_filename(string const &fname)
-{
-	filename res;
-	string::size_type idx = fname.rfind('.');
-	if(idx == std::string::npos)
-	{
-		res.valid = false;
-	}
-	else
-	{
-		res.valid = true;
-		res.name = fname.substr(0, idx);
-		res.ext = fname.substr(idx+1);
-	}
-	return res;
-}
-
 //
 // 1. Iterates through the chisel files on disk (.sc and .lua)
 // 2. Opens them and extracts the fields (name, description, etc)
 // 3. Adds them to the chisel_descs vector.
 //
 void sinsp_chisel::get_chisel_list(vector<chisel_desc>* chisel_descs)
 {
-	for(vector<chiseldir_info>::const_iterator it = g_chisel_dirs->begin();
-		it != g_chisel_dirs->end(); ++it)
+#ifdef HAS_LUA_CHISELS
+	for(auto& dir_info : *g_chisel_dirs)
 	{
-		if(string(it->m_dir).empty())
+		if(dir_info.m_dir.empty())
 		{
 			continue;
 		}
 
-		tinydir_dir dir = {};
-
-		tinydir_open(&dir, it->m_dir.c_str());
-
-		while(dir.has_next)
+		for (auto const& dir_entry : filesystem::directory_iterator(dir_info.m_dir))
 		{
-			tinydir_file file;
-			tinydir_readfile(&dir, &file);
-
-			string fpath(file.path);
-			bool add_to_vector = false;
-			chisel_desc cd;
-
-			filename fn = split_filename(string(file.name));
-			if(fn.ext != "sc" && fn.ext != "lua")
-			{
-				goto next_file;
-			}
-
-			for(vector<chisel_desc>::const_iterator it_desc = chisel_descs->begin();
-				it_desc != chisel_descs->end(); ++it_desc)
+			if(dir_entry.path().extension() == ".lua")
```

Comment:
We were accepting `sc` too:
```
if(fn.ext != "sc" && fn.ext != "lua")
```

Reply:
ah i see, we only inited lua chisel ending with lua, but eventually added to a vector the `sc` too (if asked). 
I don't get why though :)